app/models/assignment.rb

```diff
@@ -628,7 +628,8 @@ def get_detailed_csv_report
         # total percentage, total_grade
         result.concat(['','0'])
         # mark, max_mark
-        result.concat(get_criteria.pluck("''", :max_mark).flatten)
+        result.concat(Array.new(criteria_count, '').
+          zip(get_criteria(:all, :rubric).pluck(:max_mark) + get_criteria(:all, :flexible).pluck(:max_mark)).flatten)
```

Comment:
Alright, you don't need to use `pluck` here. Just map over the result of `get_criteria`